src/category_theory/groupoid/basic.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2022 Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémi Bottinelli
+-/
+import category_theory.groupoid
+import category_theory.is_connected
+import combinatorics.quiver.basic
+
+/-!
+This file defines a few basic properties of groupoids.
+-/
+
+namespace category_theory
+
+namespace groupoid
+
+variables (C : Type*) [groupoid C]
+
+section thin
+
+lemma is_thin_iff : (∀ (c d : C), subsingleton (c ⟶ d)) ↔ (∀ (c : C), subsingleton (c ⟶ c)) :=
```

Comment:
Not sure if the name is still appropriate; at least a docstring should be added to say what `thin` means.

Reply:
How's that for  a docstring?
```
/--
A quiver is thin if there is at most one arrow between any pair of objects.
For categories, this is also called being posetal.
 -/
```

Reply:
@erdOne sorry, I didn't catch that comment! Does `@[reducible] def` in the quiver file and no definition in the groupoid file (so that we must call `quiver.is_thin`) work for you? I.e. exactly as it was at some point, but with the `@[reducible]` added.

Reply:
I think that will do. Please also replace the statements in `category_theory/thin` with the new definition.